app/controllers/api/articles_controller.rb

```diff
@@ -1,19 +1,25 @@
 class Api::ArticlesController < ApplicationController
   def index
-    articles = Article.all().most_recent
+    articles = Article.all.most_recent
     if articles == []
       render json: { articles: articles }, status: 204
+    elsif params[:category]
+      articles_by_category = Article.where(category: params[:category])
+      if articles_by_category == []
+        render json: { error_message: "There is no Articles in this category" }, status: 404
+      else
+        render json: articles_by_category, each_serializer: ArticlesIndexSerializer
+      end
```

Comment:
The way I see this, this will return content of one category, right? Not the entire collection of articles grouped by categories. Is that correct?

Reply:
That is correct

Reply:
Nested if statements? My man!